deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -417,7 +424,18 @@ class WebxdcViewController: WebViewViewController {
     private func loadHtml() {
         DispatchQueue.global(qos: .userInitiated).async { [weak self] in
             guard let self else { return }
-            let url = URL(string: "\(self.INTERNALSCHEMA)://acc\(self.dcContext.id)-msg\(self.messageId).localhost/index.html")
+
+            let file = if let href {
+                if href.starts(with: "#") {
+                    "index.html" + href
+                } else {
+                    href
+                }
+            } else {
+                "index.html"
+            }
```

Comment:
I think this would be quite unintuitive to webxdc app developers reading the window.location.href. Is index.html strictly needed here? Or is it in the spec that it works like this?

Reply:
index.html  is the starting point of any webxdc, this is also documented like that at https://webxdc.org/docs/

so,  `update.href` can be `#sth` in case of single-page apps (which is the vast majority) - for multi-page apps, that needs to be `page1.html#sth`, `page2.html#sth`, `index.html#sth`

Reply:
I might be missing a bit of context but since a webserver would serve the index.html with a `window.location.href` path of just `/` (`/index.html` not needed), an implementation of webxdc using web server instead of would have a different `window.location.href` path.

So for example if you send "#sth" in a webserver implementation the path part of `window.location.href` would become "#sth" while with this implementation it would become "index.html#sth". 

https://github.com/webxdc/website/pull/90/files/bf284a1901cc5edd7f04798e4261c16fb5555600#diff-6ecb9b561809502e7de51b357752efed1f95cd22736a885e19e5eff2012cc77aR11

The new doc says "you can't assume anything about the scheme or domain part of the url" which implies you can assume the path part of the url which is not true if my example here follows the spec? Tell me if I am wrong, you have much more knowledge of webxdc.

Reply:
it is about a link to open the app, ex. if you are in a ./subpage.html file and send #foobar this doesn't mean that the link is to the current page, you have to be explicit about what page you want to open with the #fragment, because when you start a webxdc it always starts at index.html, the only situation when you can send just the #fragment is when it is for index.html, if it were always explicit it would be more obvious, but less convenient for single page apps

Reply:
e.i. the #fragment is not relative to the current open page, otherwise the behavior would be arbitrary receiving a #fragment in the current open page when it was intended for another subpage

Reply:
i changed this PR to not look into `href`, as discussed. thanks a lot